src/main/java/org/apache/commons/text/CharacterPredicates.java

```diff
@@ -48,5 +48,73 @@ public boolean test(int codePoint) {
         public boolean test(int codePoint) {
             return Character.isDigit(codePoint);
         }
-    }
+    },
+
+    /**
+     * Tests if the code points represents a number between 0 and 9.
+     *
+     * @since 1.2
+     */
+    ARABIC_NUMERALS {
+        @Override
+        public boolean test(int codePoint) {
+            return codePoint >= ZERO_CODEPOINT && codePoint <= NINE_CODEPOINT;
+        }
+    },
+
+    /**
+     * Tests if the code points represents a letter between a and z.
+     *
+     * @since 1.2
+     */
+    ASCII_LOWERCASE_LETTERS {
+        @Override
+        public boolean test(int codePoint) {
+            return codePoint >= LOWERCASE_A_CODEPOINT && codePoint <= LOWERCASE_Z_CODEPOINT;
+        }
+    },
+
+    /**
+     * Tests if the code points represents a letter between A and Z.
+     *
+     * @since 1.2
+     */
+    ASCII_UPPERCASE_LETTERS {
+        @Override
+        public boolean test(int codePoint) {
+            return codePoint >= UPPERCASE_A_CODEPOINT && codePoint <= UPPERCASE_Z_CODEPOINT;
+        }
+    },
+
+    /**
+     * Tests if the code points represents a letter between a and Z.
+     *
+     * @since 1.2
+     */
+    ASCII_LETTERS {
+        @Override
+        public boolean test(int codePoint) {
+            return ASCII_LOWERCASE_LETTERS.test(codePoint) || ASCII_UPPERCASE_LETTERS.test(codePoint);
+        }
+    },
+
+    /**
+     * Tests if the code points represents a letter between a and Z or a number between 0 and 9.
+     *
+     * @since 1.2
+     */
+    ASCII_ALPHA_NUMERALS {
+        @Override
+        public boolean test(int codePoint) {
+            return ASCII_LOWERCASE_LETTERS.test(codePoint) || ASCII_UPPERCASE_LETTERS.test(codePoint)
+                    || ARABIC_NUMERALS.test(codePoint);
+        }
+    };
+
+    private static final int NINE_CODEPOINT = 57;
```

Comment:
Wouldn't it make more sense to use an implicit widening on `'9'` to `int` rather than hardcoded constants?